rest-service/manager_rest/security/user_handler.py

```diff
@@ -102,22 +90,45 @@ def get_token_status(token):
     :param token: The token to decrypt
     :return: A tuple: (expired, invalid, user, data)
     """
-    security = current_app.extensions['security']
-    serializer = security.remember_token_serializer
-    max_age = security.token_max_age
-
     user, data, error = None, None, None
-    expired, invalid = False, False
-
-    try:
-        data = serializer.loads(token, max_age=max_age)
-    except SignatureExpired:
-        expired = True
-    except (BadSignature, TypeError, ValueError) as e:
-        invalid = True
-        error = e
-
-    if data:
-        user = user_datastore.find_user(id=data[0])
+    expired, invalid = False, True
+
+    if token.startswith('ctok-'):
+        token_parts = token.split('-')
+        if len(token_parts) == 3:
+            _, tok_id, tok_secret = token_parts
+
+            sm = get_storage_manager()
+            token = sm.get(Token, tok_id, fail_silently=True)
+
+            error = 'Unauthorized.'
```

Comment:
The full stop in error messages feels a bit weird, don't it?

Reply:
Hmm. It's also duplicating what the actual error is- I put them in because of the function needing to return them. However, I note your refactor comment, so I'll mark this as resolved and they'll likely just disappear if I do that.